### PR TITLE
Resolves Issue #13 (#35) - View Purchase Details

### PR DESCRIPTION
**Pull request by mjhcodes**
# Issue: https://github.com/the-collab-lab/tcl-4-smart-shopping-list/issues/35
## Tasks:
The main purpose of the issue was to create a way for the user to view the details of their purchase to better understand their purchase patterns. We opted to use a modal to display the item details, which is opened by clicking a button, which is found alongside each item on the list.
## Actions:
- Imported the modal, which had been created as part of a previous issue
- Created a toggle state and function for opening and closing the modal
- Created an initialized state for the modal details, as this is rendered at load time and would return an error otherwise.
- Created the hideDetails and showDetails functions, with the latter also inserting the details of the item into state
- Created the handleDetails function, which displays the four data points requested as part of the issue inside of the modal, if all four are available; the name and number of purchases are always displayed at a minimum
- Created a function to convert the seconds returned by the date purchased value into a more readable date format and incorporated this inside the modal
- Created a button to allow the opening of the modal when clicked
- Placed the modal inside the return, to be rendered if the isViewDetailOpen value is set to true
## Notes:
PR appears to mistakenly also include the changes made as part of a previous issue
## Screenshots:

**EXAMPLE 1 - VIEW DETAILS BUTTON**
Screen Shot 2020-03-07 at 4 38 58 PM

**EXAMPLE 2 - LIMITED DETAILS MODAL**
Screen Shot 2020-03-07 at 4 39 48 PM

**EXAMPLE 3 - FULL DETAILS MODAL**
Screen Shot 2020-03-07 at 4 39 14 PM

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -8,6 +8,8 @@ import getToken from "./lib/token";
 import Form from "./Form/Form";
 import * as firebase from "./lib/firebase";
 
+// heck spiky rake
+
```
Comment (rmorabia) [inline src/App.js]: Remove unnecessary comments!

**Review by rmorabia — APPROVED** [review]
LGTM.

File: src/List/ItemList.js [hunk 2]
```diff
@@ -1,13 +1,27 @@
-import React, { useState, useEffect, Fragment } from "react";
+import React, { useEffect, Fragment, useState } from "react";
 import { NavLink } from "react-router-dom";
 import "firebase/firestore";
 import * as firebase from "../lib/firebase";
-import classes from "./List.module.css";
+import Modal from "../Modal/Modal";
```
Comment (jcwesley93) [inline src/List/ItemList.js:5]: I really like that you all used Modal - I think it makes for a cleaner UI experience.

**Review by jcwesley93 — COMMENTED** [review]
I think that everything here looks good code wise. This is probably my favorite PR description thus far! I liked your use of screenshots and the detailed description. I was able to navigate the code changes easily because of this. Great work!

**Review by jcwesley93 — APPROVED** [review]

File: src/App.js (hunk 1 above)
Comment (RachaelSkye) [reply src/App.js]: LOL that was mine and I'm so sorry I keep forgetting about it and it never dies

**Comment by RachaelSkye** [comment]
Everything looks great and I definitely like using the modal to view details. Also, this has definitely been a lesson in PR requests! Thanks for being so thorough!

**Review by RachaelSkye — APPROVED** [review]

**Review by msholty-fd — APPROVED** [review]
Excellent! 🏅 :shipit:

**Pull request merged**